woocommerce/woocommerce-ios
# Order Creation: Do not prefill initial amount for new shipping line

## Description

Due to background order sync, new shipping line flow is prefilled with "0.00" amount, which is inconvenient - user need to delete it before entering something. It's only correctly empty when shipping opened as first item, without adding products to trigger first background sync.

This PR allows prefilling any initial amount only for editing existing shipping line. Creating new one will always have empty amount field.

## Testing

1. Go to the Orders tab, tap "+" in navbar and select "Create order".
2. Add a product to trigger background order sync.
3. Tap "Add Shipping".
4. Confirm that amount field is empty and allows input.
5. Enter some amount and save. Open it again.
6. Confirm that previously entered amount is correctly prefilled.
7. Repeat steps 5-6 with "0" amount which is allowed when entered manually.

## Screenshots

before|after
--|--
![Simulator Screen Shot - 1](https://user-images.githubusercontent.com/3132438/166298915-05852dac-0353-4910-991e-6fe2e36774ee.png)|![Simulator Screen Shot - 2](https://user-images.githubusercontent.com/3132438/166298920-14359c09-0484-4521-8347-678a65f650f8.png)

---
